File: app/src/main/java/com/jodelXposed/hooks/LocationStuff.java
package com.jodelXposed.hooks;

import android.location.Location;

import com.jodelXposed.utils.Options;

import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

import static de.robv.android.xposed.XposedHelpers.findAndHookMethod;

public class LocationStuff {
    public LocationStuff(XC_LoadPackage.LoadPackageParam lpparam) {
        findAndHookMethod("com.jodelapp.jodelandroidv3.data.googleservices.location.LocationManager", lpparam.classLoader, "onLocationChanged", Location.class, new XC_MethodHook() {
            @Override
            protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
                if(Options.INSTANCE.getLocation().getActive()) {
                    Location l = (Location) param.args[0];
                    l.setLatitude(Options.INSTANCE.getLocation().getLat());
                    l.setLongitude(Options.INSTANCE.getLocation().getLng());
                }
            }
        });
    }
}

File: app/src/main/java/com/jodelXposed/hooks/LocationStuff.kt
package com.jodelXposed.hooks

import android.location.Location
import com.jodelXposed.models.Hookvalues

import com.jodelXposed.utils.Options

import de.robv.android.xposed.XC_MethodHook
import de.robv.android.xposed.callbacks.XC_LoadPackage

import de.robv.android.xposed.XposedHelpers.findAndHookMethod

class LocationStuff(lpparam: XC_LoadPackage.LoadPackageParam) {
    init {
        findAndHookMethod(Options.hooks.Class_LocationChangeListener, lpparam.classLoader, "onLocationChanged", Location::class.java, object : XC_MethodHook() {
            @Throws(Throwable::class)
            override fun beforeHookedMethod(param: XC_MethodHook.MethodHookParam) {
                if (Options.location.active) {
                    val l = param.args[0] as Location
                    l.latitude = Options.location.lat
                    l.longitude = Options.location.lng
                }
            }
        })
    }
}
